pylint/utils.py

```diff
@@ -684,6 +691,12 @@ def iter_spurious_suppression_messages(self, msgs_store):
                 yield 'suppressed-message', line, \
                     (msgs_store.get_msg_display_string(warning), from_)
 
+    def get_effective_max_line_number(self):
+        """Return the maximum line number present in _module_msgs_state."""
+        if self._module_msgs_state:
+            return max([max(d.keys()) for d in self._module_msgs_state.values()])
```

Comment:
Maybe could be refactored to use generator expressions:
`max(max(d) for d in self._module_msgs_state.values))`